### PR TITLE
[Sofa.LinearAlgebra] Dispatch EigenMatrix and EigenVector from Eigen2Solver

### PR DESCRIPTION
**Pull request by fredroy**
Continuing #2150

Implementation of BaseMatrix and BaseVector using Eigen library was contained in SofaEigen2Solver.

Those EigenVector/EigenMatrix are almost only used for ad-hoc code in FEMs (getJs, geometricStiffness) related to Flexible or Compliant.

I tried to make it as compatible as possible, so it should not be breaking.

I have deleted also an accessor on a special "Eigen-"MultiMatrixAccessor:
 - it is not used
 - it would have introduced a dependency on Sofa.Core... (not possible)

Finally, I think this classes wont really be useful in the future, especially that BaseVector/BaseMatrix using Eigen are present in SofaCaribou (https://github.com/jnbrunet/caribou/blob/master/src/SofaCaribou/Algebra/EigenVector.h for example) and seems better overall.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/Sofa.LinearAlgebra/src/sofa/linearalgebra/EigenBaseSparseMatrix.h [hunk 1]
```diff
@@ -349,47 +346,47 @@ class EigenBaseSparseMatrix : public defaulttype::BaseMatrix
     }
 
 
-    /// View this matrix as a MultiMatrix
-    class MatrixAccessor: public core::behavior::MultiMatrixAccessor
-    {
-    public:
+    ///// View this matrix as a MultiMatrix
```
Comment (alxbilger) [inline SofaKernel/modules/Sofa.LinearAlgebra/src/sofa/linearalgebra/EigenBaseSparseMatrix.h]: You introduced commented code. Is is relevant?

File: SofaKernel/modules/Sofa.LinearAlgebra/src/sofa/linearalgebra/EigenSparseMatrix.h [hunk 2]
```diff
@@ -66,62 +65,58 @@ class EigenSparseMatrix : public EigenBaseSparseMatrix<typename OutDataTypes::Re
...
-	  template<class VecDeriv>
-	  static typename map_traits<VecDeriv>::map_type
-	  map( VecDeriv& data) {
-		   return map_traits<VecDeriv>::map(&data[0][0], data.size());
-	  }
-	
+
+    };
+
+    //template<class DataVecDeriv>
```
Comment (alxbilger) [inline SofaKernel/modules/Sofa.LinearAlgebra/src/sofa/linearalgebra/EigenSparseMatrix.h]: Again commented code

**Review by alxbilger — CHANGES_REQUESTED** [review]
It seems there are more changes than I expected. A description of the changes would help me to understand.

I also have the suggestion to remove the commented code.

It also worth mentioning that you add the Eigen dependency to Sofa.LinearAlgebra, which is fine for me.

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/Sofa.LinearAlgebra/src/sofa/linearalgebra/EigenVector.h [hunk 3]
```diff
@@ -20,27 +20,26 @@
...
 #include <sofa/type/Vec.h>
 #include <Eigen/Dense>
-#include <sofa/defaulttype/VecTypes.h>
 
-namespace sofa::component::linearsolver
+namespace sofa::linearalgebra
 {
 
-//#define EigenVector_CHECK
+//#define EIGEN_CHECK
```
Comment (guparan) [inline SofaKernel/modules/Sofa.LinearAlgebra/src/sofa/linearalgebra/EigenVector.h]:
```suggestion

```

**Pull request merged**